### PR TITLE
use architect orb

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -38,39 +19,19 @@ workflows:
...
           password_envar: "QUAY_PASSWORD"
           requires:
-            - build
+            - go-build
           filters:
             tags:
               only: /^v.*/
 
       - architect/push-to-docker:
-          name: push-cluster-operator-to-aliyun-pr
```
Comment (xh3b4sd) [inline .circleci/config.yml]: I made it to push to all registries all the time.

**Comment by xh3b4sd** [comment]
@kopiczko you got an idea what is going on here in the build? It checks the chart but looks like some legacy version, which sounds odd in master branch. I am also wondering why this is even validated. 

```
cluster-operator => (version: "0.23.7-0f04bcc4923f7cbe937a79cdf6c2af10dd26fc66", path: "helm/cluster-operator")
```

File: .circleci/config.yml (hunk 1 above)
Comment (kopiczko) [reply .circleci/config.yml]: Why should it be changed within this PR?

Comment (kopiczko) [reply .circleci/config.yml]: Ah yeah it was weird. I guess someone removed hold step but not merge the jobs. All good 👍

**Comment by xh3b4sd** [comment]
My question is still not answered about the weird legacy version. The linter complains about `0.23.7`, but this here is master. This does not add up for me.

**Comment by xh3b4sd** [comment]
So we seem to be good here. Rolling.

**Comment by kopiczko** [comment]
> The linter complains about `0.23.7`, but this here is master

The linter complaints about chart `0.23.7-0f04bcc4923f7cbe937a79cdf6c2af10dd26fc66`. This chart version is produced per commit and not per branch. Hope that clarifies.

**Comment by xh3b4sd** [comment]
Not really because I am still not on legacy anymore so I don't follow where `0.23.7` comes from. Latest tag from master, which the PR is based on is `v2.1.5`.

**Comment by kopiczko** [comment]
I see what you mean. I'll try to check later.

**Comment by kopiczko** [comment]
I had that on my list but since we're moving to VERSION file I won't investigate that further.

**Pull request merged**